ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -293,6 +301,27 @@ private BlockImportResult importBlockAndState(
     return result;
   }
 
+  public void updateLatestValidFinalizedSlot(final ExecutePayloadResult payloadResult) {
+    onForkChoiceThread(
+            () -> {
+              if (payloadResult.getStatus() != ExecutionPayloadStatus.VALID) {
+                return;
+              }
+              UInt64 latestValidFinalizedSlotInStore =
+                  recentChainData.getStore().getLatestFinalized().getSlot();
```

Comment:
I think there are some race conditions here.  We might get a VALID payload for block 10, but by the time we get this executed on the fork choice thread, we've imported blocks 11, 12, 13 etc which updates the finalised checkpoint to something after block 10.  We need to be passing into this method the latest finalised slot we've validated.

Reply:
right!